src/nilql/nilql.py

```diff
@@ -943,8 +948,13 @@ def decrypt(
             except Exception as exc:
                 raise error from exc
 
-        bytes_ = bytes(len(shares[0]))
-        for share_ in shares:
+        lens = [len(share) for share in shares]
+        indices = sorted(range(len(lens)), key=lambda i: -lens[i])
```

Comment:
Leave a comment here that we bring the true share first and leave the seeds last, or if everything is a share, this line doesn't do anything.